### PR TITLE
GH-143 Change compass to event based.

### PR DESCRIPTION
**Pull request by jamesmontemagno**
### Description of Change ###

Makes API for compass event based.

### Bugs Fixed ###

- Related to issue #143

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Caboodle/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Caboodle/Compass/Compass.android.cs [hunk 1]
```diff
@@ -13,10 +11,13 @@ public static partial class Compass
             Platform.SensorManager?.GetDefaultSensor(SensorType.Accelerometer) != null &&
             Platform.SensorManager?.GetDefaultSensor(SensorType.MagneticField) != null;
 
-        internal static void PlatformStart(SensorSpeed sensorSpeed, Action<CompassData> handler)
+        static SensorListener listener;
+        static Sensor magnetometer;
+        static Sensor accelerometer;
```
Comment (Redth) [inline Caboodle/Compass/Compass.android.cs]: Can we dispose/null these out when they are no longer needed?

Comment (jamesmontemagno) [reply Caboodle/Compass/Compass.android.cs]: so they are shared system wide and if you dispose them out then they will effect other listeners from my testing :(

Comment (jamesmontemagno) [reply Caboodle/Compass/Compass.android.cs]: Well actually.... maybe if we only allow one at a time.... let me see here. Will need to see if you can do both compass and others at the same time.

Comment (Redth) [reply Caboodle/Compass/Compass.android.cs]: I guess it could get ugly, but could we track/count the # of listeners we have and dispose if 0?

Comment (mattleibow) [reply Caboodle/Compass/Compass.android.cs]: What if we use a weak reference? This way we leave it up the Androids?

Comment (jamesmontemagno) [reply Caboodle/Compass/Compass.android.cs]: #resolved

**Pull request merged**